R/gateUtil.R

```diff
@@ -89,7 +89,7 @@ commonGateCreate = function(body, name, gid,
                             experimentId,
                             parentPopulationId, parentPopulation,
                             tailoredPerFile, fcsFileId, fcsFile,
-                            createPopulation) {
+                            createPopulation, gids = NULL) {
```

Comment:
My R is rusty. Where this function is used, `gids` is passed as the 4th argument. Does R do implicit named parameter matching (variable's name is used for parameter name matching, and the same variable's value is used)? I know it does full name, partial name and positional matching.

Reply:
"Arguments are matched first by exact name (perfect matching), then by prefix matching, and finally by position. "
http://adv-r.had.co.nz/Functions.html

It works, but it isn't syntactically very clean. Happy to shuffle things around to be positionally correct also.

Reply:
I was wondering about how you're calling it, like `compoundGateCreate(body, names, gid, gids, experimentId, parentPopulationId, parentPopulation, tailoredPerFile, fcsFileId, fcsFile, createPopulation)` -- is that like shorthand for `compoundGateCreate(body, names, gid, gids=gids, .....)` (with the equals sign)?

Reply:
Oh, funny, I see what happened. I initially tried to modify `commonGateCreate` to work for both compound and normal gates, and the `gids = NULL` was a holdover that I forgot to delete. 

Regarding your question, I am calling `gids` positionally (which is kind of a shorthand for `gids=gids`). You'll note that `commonGateCreate` has a final arg `gids = NULL` but `compoundGateCreate` has the args `(..., gid, gids, ...)`